notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,512 @@
+mnl_specification_causal_2 = mnl_specification_noncausal.copy()
+mnl_names_causal_2 = mnl_names_noncausal.copy()
+
+mnl_specification_causal_2["confounder_all_2"] = [1, 2, 3]
+mnl_names_causal_2["confounder_all_2"] = ["Confounder - Drive alone",
+                                     "Confounder - Shared ride 2", 
+                                     "Confounder - Shared ride 3"]
+
+# +
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model_causal_2 = cm.create_choice_model(data=data,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_causal_2,
+                                   model_type="MNL",
+                                   names=mnl_names_causal_2)
+
+num_vars = len(reduce(lambda x, y: x + y, mnl_names_causal_2.values()))
+# Note newton-cg used to ensure convergence to a point where gradient 
+# is essentially zero for all dimensions. 
+mnl_model_causal_2.fit_mle(np.zeros(num_vars),
+                  method="BFGS")
+
+# Look at the estimation results
+mnl_model_causal_2.get_statsmodels_summary()
+# -
+
+# ## Compare estimates on travel time and cost
+
+# +
+results_summary_true = mnl_model.get_statsmodels_summary()
+results_summary_non_causal = mnl_model_noncausal.get_statsmodels_summary()
+results_summary_method_1 = mnl_model_causal_1.get_statsmodels_summary()
+results_summary_method_2 = mnl_model_causal_2.get_statsmodels_summary()
+
+results_as_html_true = results_summary_true.tables[1].as_html()
+results_as_html_true = pd.read_html(results_as_html_true, header=0, index_col=0)[0]
+
+results_as_html_noncausal = results_summary_non_causal.tables[1].as_html()
+results_as_html_noncausal = pd.read_html(results_as_html_noncausal, header=0, index_col=0)[0]
+
+results_as_html_method_1 = results_summary_method_1.tables[1].as_html()
+results_as_html_method_1 = pd.read_html(results_as_html_method_1, header=0, index_col=0)[0]
+
+results_as_html_method_2 = results_summary_method_2.tables[1].as_html()
+results_as_html_method_2 = pd.read_html(results_as_html_method_2, header=0, index_col=0)[0]
+
+# +
+locs = ['Travel Time, units:min (Drive Alone)','Travel Time, units:min (SharedRide-2)',
+       'Travel Time, units:min (SharedRide-3+)', 'Travel Cost, units:$ (Drive Alone)',
+       'Travel Cost, units:$ (SharedRide-2)', 'Travel Cost, units:$ (SharedRide-3+)']
+cols = ['coef', 'std err']
+
+results_comparison = results_as_html_true.loc[locs][cols].join(results_as_html_noncausal.loc[locs][cols], 
+                                                               lsuffix = '_true', rsuffix = '_non_causal').join(
+    results_as_html_method_1.loc[locs][cols].join(results_as_html_method_2.loc[locs][cols],
+                                               lsuffix = '_method_1', rsuffix = '_method_2'))
+
+results_comparison
```

Comment:
I think this is one place where a plot is worth a thousand words. If you made some grouped bar plots with error bars, I imagine it would be much easier to see your point as compared to showing just the table.

You could use https://seaborn.pydata.org/generated/seaborn.barplot.html with a "long-format" dataframe where the hue is the model name (Model 1, Model 2, Model 3), the x is the various coefficient name (e.g., `Travel Cost, units:$ (Drive Alone)`), and y is the coef value. Each row of the dataframe should be a single coefficient for a single model (not all models on the same row as in `results_comparison`).